src/runtime/vdom/vdom-render.ts

```diff
@@ -204,10 +204,19 @@ const relocateToHostRoot = (parentElm: Element) => {
   plt.$flags$ &= ~PLATFORM_FLAGS.isTmpDisconnected;
 };
 
-const putBackInOriginalLocation = (parentElm: Node, recursive: boolean) => {
+const putBackInOriginalLocation = (parentElm: d.RenderNode, recursive: boolean) => {
   plt.$flags$ |= PLATFORM_FLAGS.isTmpDisconnected;
+  const oldSlotChildNodes: ChildNode[] = Array.from(parentElm.childNodes);
+
+  if (parentElm['s-sr']) {
+    let node = parentElm;
+    while ((node = node.nextSibling as d.RenderNode)) {
+      if (node && node['s-sn'] === parentElm['s-sn'] && node['s-sh'] === hostTagName) {
+        oldSlotChildNodes.push(node);
+      }
+    }
+  }
```

Comment:
Let's put this behind the `experimentalSlotFixes` bug to avoid chances that this impacts behavior of current users that are not opt-into this.
```suggestion
  if (parentElm['s-sr'] && BUILD.experimentalSlotFixes) {
    let node = parentElm;
    while ((node = node.nextSibling as d.RenderNode)) {
      if (node && node['s-sn'] === parentElm['s-sn'] && node['s-sh'] === hostTagName) {
        oldSlotChildNodes.push(node);
      }
    }
  }
```

Reply:
done 🕺